GMG/Scenes/ChordProgress/ChordProgressView.swift

```diff
@@ -347,16 +347,40 @@ extension ChordProgressView {
         }
 
         struct ToggleButton: View {
-            let title: String
+            let title: Text
             let isSelected: Bool
             let namespace: Namespace.ID
             let action: () -> Void
 
+            init<S: StringProtocol>(
+                title: S,
+                isSelected: Bool,
+                namespace: Namespace.ID,
+                action: @escaping () -> Void
+            ) {
+                self.title = Text(title)
+                self.isSelected = isSelected
+                self.namespace = namespace
+                self.action = action
+            }
```

Comment:
제네릭 타입을 사용하지 않아도 되는 상황 같습니다. 다른 코드와 통일성을 고려한다면 빼는게 좋을 것 같은데 어떻게 생각하시나요?
```suggestion
            init(
                title: StringProtocol,
                isSelected: Bool,
                namespace: Namespace.ID,
                action: @escaping () -> Void
            ) {
                self.title = Text(title)
                self.isSelected = isSelected
                self.namespace = namespace
                self.action = action
            }
```

Reply:
SwiftUI [Text.init(_:)](https://developer.apple.com/documentation/swiftui/text/init(_:)-9d1g4)이 이런 식으로 구현되어 있어서 왜 그런지 찾아봤더니 제네릭을 사용하면 컴파일러가 `StringProtocol`을 사용하는 구체 타입(`String`, `Substring`)별로 최적화를 할 수 있어서 성능상 이점이 있다고 합니다. 프로토콜을 사용할 떄는 동적 디스패치를 한번 거쳐야 되기 때문에 비교적 성능상 불리할 수 있다고 합니다.